Dockerfile

```diff
@@ -16,61 +25,82 @@ RUN turbo prune --scope=@kampus-apps/gql --docker
+# pasaport-builder stage, use "turbo" to build the project incrementally and get incremental builds
+FROM node-slim-image AS pasaport-builder
 WORKDIR /app
 COPY .gitignore .gitignore
 COPY --from=pasaport-prune /app/out/json .
 COPY --from=pasaport-prune /app/out/package-lock.json ./package-lock.json
 COPY /tsconfig.json tsconfig.json
 RUN npm install
 COPY --from=pasaport-prune /app/out/full/ .
+COPY .env ./apps/pasaport/.env
 COPY turbo.json turbo.json
 RUN npx turbo build
 
 # Use distroless to reduce the minimum size of the image and reduce the attack surface
 #If you want to use npm, use the node:18-bullseye-slim image (prisma doesn't work with alpine arm64)
 # kampus stage - the dev server.
-FROM gcr.io/distroless/nodejs18-debian11 as kampus
-ARG NODE_ENV=development
-COPY --from=kampus-builder /app .
-EXPOSE 3000
-CMD ["node_modules/next/dist/bin/next", "dev", "-p", "3000"]
-#CMD ["npm", "run", "dev", "-w", "@kampus-apps/kampus"]
+FROM node-distroless-image as kampus
+COPY tsconfig.json ./tsconfig.json
+COPY --from=kampus-builder /app/apps/kampus/next.config.mjs .
+COPY --from=kampus-builder /app/apps/kampus/package.json .
+COPY --from=kampus-builder /app/apps/kampus/.next/standalone ./
+COPY --from=kampus-builder /app/apps/kampus/.next/static ./apps/kampus/.next/static
+COPY --from=kampus-builder /app/apps/kampus/public ./apps/kampus/public
+CMD ["apps/kampus/server.js"]
 
 # gql stage - the dev server.
-FROM gcr.io/distroless/nodejs18-debian11 as gql
-COPY --from=gql-builder /app .
-EXPOSE 3002
-CMD ["node_modules/next/dist/bin/next", "dev", "-p", "4000"]
-#CMD ["npm", "run", "dev", "-w", "@kampus-apps/gql"]
+FROM node-distroless-image as gql
+COPY tsconfig.json ./tsconfig.json
+COPY --from=gql-builder /app/apps/gql/next.config.mjs .
+COPY --from=gql-builder /app/apps/gql/package.json .
+COPY --from=gql-builder /app/apps/gql/.next/standalone ./
+COPY --from=gql-builder /app/apps/gql/.next/static ./apps/gql/.next/static
+CMD ["apps/gql/server.js"]
 
 # pasaport stage - the dev server.
-FROM gcr.io/distroless/nodejs18-debian11 as pasaport
-COPY --from=pasaport-builder /app .
-EXPOSE 3001
-CMD ["node_modules/next/dist/bin/next", "dev", "-p", "3001"]
-#CMD ["npm", "run", "dev", "-w", "@kampus-apps/pasaport"]
+FROM node-distroless-image as pasaport
+COPY tsconfig.json ./tsconfig.json
+COPY --from=pasaport-builder /app/apps/pasaport/next.config.mjs .
+COPY --from=pasaport-builder /app/apps/pasaport/package.json .
+COPY --from=pasaport-builder /app/apps/pasaport/.next/standalone ./
+COPY --from=pasaport-builder /app/apps/pasaport/.next/static ./apps/pasaport/.next/static
+CMD ["apps/pasaport/server.js"]
+
+# ssl-proxy stage - the dev server.
+FROM node:18-bullseye-slim as proxy
+WORKDIR /app
+
+RUN npm i -g local-ssl-proxy
+RUN npm i -g concurrently
 
+# keep running
+CMD ["concurrently", "local-ssl-proxy --target 3000 --source 5000", "local-ssl-proxy --target 3001 --source 5001", "local-ssl-proxy --target 4000 --source 5002"]
```

Comment:
can turbo run these ? turbo has concurrent running capability

Reply:
at this point we don't need turbo, we pruned everything, and i am not even sure if there is a turbo binary at this point. i think it's better to go with this.